packages/migrate/ResolverIntercept.js

```diff
@@ -0,0 +1,39 @@
+class ResolverIntercept {
+  constructor(resolver) {
+    this.resolver = resolver;
+    this.cache = [];
+  }
+
+  require(contractName) {
+    // remove file extension if present on name
+    const sanitizedContractName = contractName
+      .replace(/^\.\//, "")
+      .replace(/\.sol$/i, "");
+
+    // there may be more than one contract of the same name which will be
+    // problematic - only return the first one found in the cache for now
+    for (const contract of this.cache) {
+      if (contract.contract_name === sanitizedContractName) {
```

Comment:
do we need to handle `contract.contractName`? I know that in some places in the case, there's risk of either casing being used. Not sure if this is one of them.

Reply:
I think this will be Truffle artifact objects so I believe we don't need to handle that case.